hail/src/main/scala/is/hail/expr/types/encoded/EArray.scala

```diff
@@ -0,0 +1,62 @@
+package is.hail.expr.types.encoded
+
+import is.hail.annotations.Region
+import is.hail.asm4s._
+import is.hail.expr.types.BaseType
+import is.hail.expr.types.physical._
+import is.hail.expr.types.virtual._
+import is.hail.io.{InputBuffer, OutputBuffer}
+import is.hail.utils._
+
+final case class EArray(elementType: EType, override val required: Boolean = false) extends EType {
+  lazy val virtualType: TArray = TArray(elementType.virtualType, required)
+
+  def compatible(pt: PType): Boolean = {
```

Comment:
would suggest another indirection with compatible too -- have a `compatible` that takes the fundamental type, and calls `_compatible`

Reply:
Good point, This is about to get weird though because compatibility relations when it comes to encoders and decoders are basically converses.